src/components/Detail/UpdateForm.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState } from 'react';
+
+import { trpc } from '../../utils/trpc';
+import { Button } from '../Button/Button';
+import { TextField } from '../TextField/TextField';
+import style from './UpdateForm.module.scss';
+
+type UpdateFormProps = {
+  activity: {
+    name: string;
+    amount: number;
+    unit: string;
+    timesDone: number;
+    id: number;
```

Comment:
The types could be imported from Prisma (or Trpc). So you don't need to write them again. That is the whole power of prisma-trpc setup.

Reply:
https://github.com/prisma/prisma/discussions/12453#discussioncomment-2410437

Reply:
did not realise